packages/offix-cache/src/createMutationOptions.ts

```diff
@@ -128,94 +133,101 @@ export const createMutationOptions = <T = {
 
 /**
  * Generate the update function to update the cache for a given operation and query.
-
  **/
-export const getUpdateFunction = (options: CacheUpdateOptions): MutationUpdaterFn => {
+const getUpdateFunction = (options: CacheUpdateOptions): MutationUpdaterFn => {
   if (!options.updateQuery) {
     throw new Error("Required updateQuery parameter is not supplied");
   }
   if (!options.mutationName) {
     throw new Error("Required mutationName parameter is not supplied");
   }
-  const {
-    mutationName,
-    updateQuery,
-    operationType = CacheOperation.ADD,
-    idField = "id"
-  } = options;
 
-  const { query, variables } = deconstructQuery(updateQuery);
-  const queryField = getOperationFieldName(query);
-  let updateFunction: MutationUpdaterFn;
+  const { operationType = CacheOperation.ADD, ...cacheHelperOptions } = options
+
   if (operationType === CacheOperation.ADD) {
-    updateFunction = (cache, { data }) => {
-      let queryResult;
-      if (data) {
-        const operationData = data[mutationName];
-        try {
-          queryResult = cache.readQuery({ query, variables }) as any;
-        } catch (e) {
-          queryResult = {};
+   return addItemToQuery(cacheHelperOptions);
+  } 
+  if (operationType === CacheOperation.DELETE) {
+    return deleteItemFromQuery(cacheHelperOptions);
+  } 
+  // this default catches the REFRESH case and returns an empty update function which does nothing
+  return () => {};
+};
+
+/**
+ * Generic cache update function that adds an item to a query that contains a list of items
+ * Might be exported in the future
+ */
+function addItemToQuery({ mutationName, updateQuery, idField = "id" }: CacheUpdateHelperOptions): MutationUpdaterFn {
```

Comment:
Lets export this function

Reply:
Lets rename it to `addResultToCache` or `addObjectToQuery`. Item sounds to specific

Reply:
`addResultToQuery` sounds like a nice one